dts/xtensa/nxp/nxp_imx8m.dtsi

```diff
@@ -17,6 +17,16 @@
 			device_type = "cpu";
 			compatible = "cdns,tensilica-xtensa-lx6";
 			reg = <0>;
+
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			clic: interrupt-controller@0 {
+				compatible = "cdns,xtensa-core-intc";
```

Comment:
Looks like no SoC is using `nxp,irqsteer-intc` compatible, can you clarify who will use the new interrupt controller driver

Reply:
i.MX8QM, QXP and MP. We can't enable it at the moment as it would break SOF. See example above of how enabling it would look like on MP.

Reply:
Sorry I don't see the example. I am trying to understand why this driver is needed if no one is using it.

Reply:
Because it's going to become a requirement later on when we switch SOF to using Zephyr native drivers on QM, QXP, and MP. I've decided to introduce the driver right now (despite knowing we can't enable it on any of said platforms ATM) instead of later on as it has no dependencies and it would ease the review process and save us some time.

Reply:
@mmahadevan108 there was also a discussion [here](https://github.com/zephyrproject-rtos/zephyr/pull/64637#discussion_r1397363785), about enabling UART interrupts on i.MX8MP, which will use irq-steer.